src/server/rest/v1/service/AssetService.ts

```diff
@@ -321,35 +271,32 @@ export default class AssetService {
     // Check if component is active
     UtilsService.assertComponentIsActiveFromToken(req.user, TenantComponents.ASSET,
       Action.LIST, Entity.ASSET, MODULE_NAME, 'handleGetAssetsInError');
-    // Check auth
-    if (!await Authorizations.canListAssetsInError(req.user)) {
-      throw new AppAuthError({
-        errorCode: HTTPAuthError.FORBIDDEN,
-        user: req.user,
-        action: Action.IN_ERROR, entity: Entity.ASSET,
-        module: MODULE_NAME, method: 'handleGetAssetsInError'
-      });
-    }
     // Filter
     const filteredRequest = AssetValidator.getInstance().validateAssetsGetReq(req.query);
-    // Build error type
-    const errorType = (filteredRequest.ErrorType ? filteredRequest.ErrorType.split('|') : [AssetInErrorType.MISSING_SITE_AREA]);
+    // Check dynamic auth
+    const authorizationAssetsInErrorFilter = await AuthorizationService.checkAndGetAssetsAuthorizations(
+      req.tenant, req.user, Action.IN_ERROR, filteredRequest);
```

Comment:
I would expect this code just after (cf. SiteService endpoint):
```
    if (! authorizationAssetsInErrorFilter.authorized) {
      UtilsService.sendEmptyDataResult(res, next);
      return;
    }
```

Reply:
The checkAndGetAssetsAuthorizations will throw an exception if the auth fails, I set the failsWithException = true